discojs/discojs-core/src/validation/validator.ts

```diff
@@ -18,32 +18,48 @@ export class Validator {
     }
   }
 
-  async assess (data: data.Data): Promise<void> {
+  private getLabel (ys: tf.Tensor): Float32Array | Int32Array | Uint8Array {
+    if (this.task.trainingInformation.modelCompileData.loss === 'binaryCrossentropy') {
+      return ys.greaterEqual(tf.scalar(0.5)).dataSync()
+    } else {
+      return ys.argMax(1).dataSync()
+    }
+  }
+
+  async assess (data: data.Data): Promise<Array<{groundTruth: number, pred: number, features: number | number[] | number[][] | number[][][] | number[][][][] | number[][][][][]}>> {
     const batchSize = this.task.trainingInformation?.batchSize
     if (batchSize === undefined) {
       throw new TypeError('batch size is undefined')
     }
 
-    const labels: string[] | undefined = this.task.trainingInformation?.LABEL_LIST
-    const classes = labels?.length ?? 1
-
     const model = await this.getModel()
 
+    let features: Array<number | number[] | number[][] | number[][][] | number[][][][] | number[][][][][]> = []
```

Comment:
the `xs` vector is always a batch of samples, hence we usually directly cast it as `number[][]` to avoid carrying TF.js's bad typing